libs/wire-api/src/Wire/API/Meeting.hs

```diff
@@ -0,0 +1,133 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2025 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Wire.API.Meeting where
+
+import Control.Lens ((?~))
+import Data.Aeson ()
+import Data.Id (ConvId, UserId, uuidSchema)
+import Data.Json.Util (utcTimeSchema)
+import Data.OpenApi qualified as S
+import Data.Qualified (Qualified, qualifiedSchema)
+import Data.Schema
+import Data.Time.Clock
+import Data.UUID (UUID)
+import Deriving.Aeson
+import Imports
+import Servant (FromHttpApiData, ToHttpApiData)
+import Wire.API.User.Identity (EmailAddress)
+import Wire.Arbitrary (Arbitrary, GenericUniform (..))
+
+-- | Unique identifier for a meeting
+newtype MeetingId = MeetingId {unMeetingId :: UUID}
+  deriving stock (Eq, Ord, Show, Generic)
+  deriving newtype (ToJSON, FromJSON, FromHttpApiData, ToHttpApiData, S.ToSchema, S.ToParamSchema)
+  deriving (Arbitrary) via (GenericUniform MeetingId)
+
+instance ToSchema MeetingId where
+  schema = MeetingId <$> unMeetingId .= uuidSchema
+
+instance ToSchema (Qualified MeetingId) where
+  schema = qualifiedSchema "MeetingId" "id" schema
+
+-- | Core Meeting type
+data Meeting = Meeting
+  { id :: Qualified MeetingId,
+    title :: Text,
+    creator :: UserId,
+    startDate :: UTCTime,
+    endDate :: UTCTime,
+    schedule :: Maybe Text,
```

Comment:
IMO the cron format is a mistake here, there is some discussion about it in the RFC. We should wait for that to resolve.